File: List-ADT-Operations/list_adt.py
class ListADT:
    def __init__(self, capacity=10):
        self.elements = []
        self.capacity = capacity

    # Insert element at the head
    def insert_at_head(self, element):
        if len(self.elements) == self.capacity:
            print("List is full")
            return
        self.elements.insert(0, element)

    # Print the list
    def print_list(self):
        for item in self.elements:
            print(item, end=" ")
        print()

# Main execution
if __name__ == "__main__":
    list_adt = ListADT()

    list_adt.insert_at_head(10)
    list_adt.insert_at_head(20)
    list_adt.insert_at_head(30)

    list_adt.print_list()

File: List-ADT-Operations/ListADT.java
import java.util.Arrays;

class ListADT {
    private int[] elements;
    private int size;

    public ListADT(int capacity) {
        elements = new int[capacity];
        size = 0;
    }

    // Insert element at the head
    public void insertAtHead(int element) {
        if (size == elements.length) {
            System.out.println("List is full");
            return;
        }
        for (int i = size - 1; i >= 0; i--) {
            elements[i + 1] = elements[i];
        }
        elements[0] = element;
        size++;
    }

    // Print the list
    public void printList() {
        for (int i = 0; i < size; i++) {
            System.out.print(elements[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        ListADT list = new ListADT(10);

        list.insertAtHead(10);
        list.insertAtHead(20);
        list.insertAtHead(30);

        list.printList();
    }
}
